tests/ImageKit/Upload/UploadTest.php

```diff
@@ -60,144 +67,330 @@ private function stubHttpClient($response)
+                [
+                    "name" => "remove-bg",
+                    "options" => [  // all parameters inside this object are sent directly to the third-party service
+                        "add_shadow" => true
+                    ]
+                ]
+            ],
+            "webhookUrl" => "https://example.com/webhook",                      // Notification URL to receive the final status of pending extensions
+            "overwriteFile" => true,                                            // true|false, in case of false useUniqueFileName should be true
+            "overwriteAITags" => false,                                          // true|false, set to false in order to preserve overwriteAITags
+            "overwriteTags" => false,                                            // true|false
+            "overwriteCustomMetadata" => true,                                  // true|false
+            "customMetadata" => [                                              // An array of created custom fields, for more details refer to docs
+                    "SKU" => "VS882HJ2JD",
+                    "price" => 599.99,
+            ]
+        ];
+
+        $mockBodyResponse = Utils::streamFor(json_encode($this->uploadSuccessResponseObj));
+
+
+        $mock = new MockHandler([
+            new Response(200, ['X-Foo' => 'Bar'], $mockBodyResponse)
+        ]);
 
-        $this->stubHttpClient(new Response(200, ['X-Foo' => 'Bar'], $mockBodyResponse));
+        $handlerStack = HandlerStack::create($mock);
 
-        $res = $this->client->upload([
-            'file' => 'http://lorempixel.com/640/480/',
-            'fileName' => '7e57d004-2b97-0e7a-b45f-5387367791cd',
+        $container = [];
+        $history = Middleware::history($container);
+
+        $handlerStack->push($history);
+        
+        $mockClient = new ImageKit(
+            'testing_public_key',
+            'testing_private_key',
+            'https://ik.imagekit.io/demo',
+            Transformation::DEFAULT_TRANSFORMATION_POSITION,
+            $handlerStack
+        );
+        
+        $response = $mockClient->uploadFile($fileOptions);
+        
+        $requestBody = $container[0]['request']->getBody();
+        $requestHeaders = $container[0]['request']->getHeaders();
+        $boundary = str_replace("multipart/form-data; boundary=","",$requestHeaders["Content-Type"][0]);
+
+        UploadTest::assertArrayHasKey("Content-Type",$requestHeaders);
+        UploadTest::assertStringStartsWith("multipart/form-data; boundary=",$requestHeaders['Content-Type'][0]);
+
+        $stream = Utils::streamFor($requestBody);
+        $stream = str_replace('\r\n',' ',json_encode($stream->getContents()));
+
+        $this->checkFormData($stream,$boundary,"file",$fileOptions['file']);
+        $this->checkFormData($stream,$boundary,"fileName",$fileOptions['fileName']);
+        $this->checkFormData($stream,$boundary,"tags",implode(',',["abd", "def"]));
```

Comment:
Instead of writing `implode(',',["abd", "def"])` here write `abd,def` explicityly like you have done in other test case